code/espurna/homeassistant.ino

```diff
@@ -31,17 +31,21 @@ String _haFixName(String name) {
 
 #if SENSOR_SUPPORT
 
-void _haSendMagnitude(unsigned char i, JsonObject& config) {
+void _haSendMagnitude(unsigned char i, const JsonObject& deviceConfig, JsonObject& config) {
 
     unsigned char type = magnitudeType(i);
     config["name"] = _haFixName(getSetting("hostname") + String(" ") + magnitudeTopic(type));
     config.set("platform", "mqtt");
     config["state_topic"] = mqttTopic(magnitudeTopicIndex(i).c_str(), false);
     config["unit_of_measurement"] = magnitudeUnits(type);
 
+    if (deviceConfig.size() > 0) {
+        config["uniq_id"] = getIdentifier() + "_" + type + "_" + String(i);
```

Comment:
Note that this uses internal type id. I think this should be `magnitudeTopic(type)`

Reply:
the unique_id is an ID that is not supposed to be user friendly, so I think the magnitudeType char is ok for this.

Reply:
I mean, it is internal to the espurna, and might be changed to be something else. String is unlikely to change though.

Reply:
Oh ok. I thought that the string was more likely to change. I'll changed that then.